File: src/main/java/Models/JobOffer.java
package Models;

import java.sql.Date;

public class JobOffer {

    private int jobOfferId;
    private String title;
    private String description;
    private String contractType;
    private double salary;
    private String location;
    private int experienceRequired;
    private Date publicationDate;
    private String status;
    private int user_id;  // new field

    // ======== DEFAULT CONSTRUCTOR ========
    /**
     * Creates a new JobOffer instance.
     */
    public JobOffer() {
    }

    // ======== PARAMETERIZED CONSTRUCTOR ========
    public JobOffer(String title, String description, String contractType,
                    double salary, String location, int experienceRequired,
                    Date publicationDate, String status, int user_id) {
        this.title = title;
        this.description = description;
        this.contractType = contractType;
        this.salary = salary;
        this.location = location;
        this.experienceRequired = experienceRequired;
        this.publicationDate = publicationDate;
        this.status = status;
        this.user_id = user_id;
    }
    public JobOffer(int jobOfferId, String title, String description, String contractType,
                    double salary, String location, int experienceRequired,
                    Date publicationDate, String status, int user_id) {

        this.jobOfferId = jobOfferId;
        this.title = title;
        this.description = description;
        this.contractType = contractType;
        this.salary = salary;
        this.location = location;
        this.experienceRequired = experienceRequired;
        this.publicationDate = publicationDate;
        this.status = status;
        this.user_id = user_id;
    }

    // ======== GETTERS & SETTERS ========
    /**
     * Returns the jobofferid value.
     */
    public int getJobOfferId() {
        return jobOfferId;
    }

    /**
     * Sets the jobofferid value.
     */
    public void setJobOfferId(int jobOfferId) {
        this.jobOfferId = jobOfferId;
    }

    /**
     * Returns the title value.
     */
    public String getTitle() {
        return title;
    }

    /**
     * Sets the title value.
     */
    public void setTitle(String title) {
        this.title = title;
    }

    /**
     * Returns the description value.
     */
    public String getDescription() {
        return description;
    }

    /**
     * Sets the description value.
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * Returns the contracttype value.
     */
    public String getContractType() {
        return contractType;
    }

    /**
     * Sets the contracttype value.
     */
    public void setContractType(String contractType) {
        this.contractType = contractType;
    }

    /**
     * Returns the salary value.
     */
    public double getSalary() {
        return salary;
    }

    /**
     * Sets the salary value.
     */
    public void setSalary(double salary) {
        this.salary = salary;
    }

    /**
     * Returns the location value.
     */
    public String getLocation() {
        return location;
    }

    /**
     * Sets the location value.
     */
    public void setLocation(String location) {
        this.location = location;
    }

    /**
     * Returns the experiencerequired value.
     */
    public int getExperienceRequired() {
        return experienceRequired;
    }

    /**
     * Sets the experiencerequired value.
     */
    public void setExperienceRequired(int experienceRequired) {
        this.experienceRequired = experienceRequired;
    }

    /**
     * Returns the publicationdate value.
     */
    public Date getPublicationDate() {
        return publicationDate;
    }

    /**
     * Sets the publicationdate value.
     */
    public void setPublicationDate(Date publicationDate) {
        this.publicationDate = publicationDate;
    }

    /**
     * Returns the status value.
     */
    public String getStatus() {
        return status;
    }

    /**
     * Sets the status value.
     */
    public void setStatus(String status) {
        this.status = status;
    }

    /**
     * Returns the user_id value.
     */
    public int getUser_id() {
        return user_id;
    }

    /**
     * Sets the user_id value.
     */
    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }
}

File: src/Entity/Joboffer.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\JobofferRepository;

#[ORM\Entity(repositoryClass: JobofferRepository::class)]
#[ORM\Table(name: 'joboffer')]
class Joboffer
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $jobOfferId = null;

    public function getJobOfferId(): ?int
    {
        return $this->jobOfferId;
    }

    public function setJobOfferId(int $jobOfferId): self
    {
        $this->jobOfferId = $jobOfferId;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $title = null;

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(?string $title): self
    {
        $this->title = $title;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $contractType = null;

    public function getContractType(): ?string
    {
        return $this->contractType;
    }

    public function setContractType(?string $contractType): self
    {
        $this->contractType = $contractType;
        return $this;
    }

    #[ORM\Column(type: 'decimal', nullable: true)]
    private ?float $salary = null;

    public function getSalary(): ?float
    {
        return $this->salary;
    }

    public function setSalary(?float $salary): self
    {
        $this->salary = $salary;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $location = null;

    public function getLocation(): ?string
    {
        return $this->location;
    }

    public function setLocation(?string $location): self
    {
        $this->location = $location;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $experienceRequired = null;

    public function getExperienceRequired(): ?int
    {
        return $this->experienceRequired;
    }

    public function setExperienceRequired(?int $experienceRequired): self
    {
        $this->experienceRequired = $experienceRequired;
        return $this;
    }

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $publicationDate = null;

    public function getPublicationDate(): ?\DateTimeInterface
    {
        return $this->publicationDate;
    }

    public function setPublicationDate(?\DateTimeInterface $publicationDate): self
    {
        $this->publicationDate = $publicationDate;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $status = null;

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(?string $status): self
    {
        $this->status = $status;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'joboffers')]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'user_id')]
    private ?User $user = null;

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;
        return $this;
    }

    #[ORM\OneToMany(targetEntity: Application::class, mappedBy: 'joboffer')]
    private Collection $applications;

    public function __construct()
    {
        $this->applications = new ArrayCollection();
    }

    /**
     * @return Collection<int, Application>
     */
    public function getApplications(): Collection
    {
        if (!$this->applications instanceof Collection) {
            $this->applications = new ArrayCollection();
        }
        return $this->applications;
    }

    public function addApplication(Application $application): self
    {
        if (!$this->getApplications()->contains($application)) {
            $this->getApplications()->add($application);
        }
        return $this;
    }

    public function removeApplication(Application $application): self
    {
        $this->getApplications()->removeElement($application);
        return $this;
    }

}
